### PR TITLE
fix: Adding a permission check for creating new datasource on generate page

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Adding a permission check for creating new datasource on generate page. When a custom role is assigned in GAC, where the user doesn't have access to create a new datasource, in spite of that the create new datasource option is showed in the dropdown on generate page with data. This is fixed with this PR.

Fixes [#22027](https://github.com/appsmithorg/appsmith/issues/22027)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx [hunk 1]
```diff
@@ -74,6 +74,8 @@ import {
 } from "selectors/onboardingSelectors";
 import { datasourcesEditorIdURL, integrationEditorURL } from "RouteBuilder";
 import { PluginPackageName } from "entities/Action";
+import { getCurrentAppWorkspace } from "@appsmith/selectors/workspaceSelectors";
+import { hasCreateDatasourcePermission } from "ce/utils/permissionHelpers";
```
Comment (berzerkeer) [inline app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx:78]: Shouldn't this be from `@appsmith`

**Review by berzerkeer — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GeneratePageForm.tsx]: oh yes. Thanks for pointing it out.

**Review by berzerkeer — APPROVED** [review]

**Review by dipyamanbiswas07 — APPROVED** [review]

**Pull request merged**